sqlglot/expressions.py

```diff
@@ -3214,10 +3214,18 @@ def parts(self) -> t.List[Expression]:
 
     def to_column(self, copy: bool = True) -> Alias | Column | Dot:
         parts = self.parts
-        col = column(*reversed(parts[0:4]), fields=parts[4:], copy=copy)  # type: ignore
+        last_part = parts[-1]
+
+        if isinstance(last_part, Identifier):
+            col = column(*reversed(parts[0:4]), fields=parts[4:], copy=copy)  # type: ignore
+        else:
+            # This branch will be reached if a function or array is wrapped in a `Table`
+            col = last_part
```

Comment:
What will `last_part` be in this case?

Reply:
Could be a `Func` instance (`SELECT * FROM ... LEFT ARRAY JOIN FOO() ...`), or an `Array`, e.g.:

```python
>>> from sqlglot import parse_one
>>> parse_one("select * from t left array join [1, 2, 3] as arr", "clickhouse")
Select(
  expressions=[
    Star()],
  from=From(
    this=Table(
      this=Identifier(this=t, quoted=False))),
  joins=[
    Join(
      this=Table(
        this=Array(
          expressions=[
            Literal(this=1, is_string=False),
            Literal(this=2, is_string=False),
            Literal(this=3, is_string=False)]),
        alias=TableAlias(
          this=Identifier(this=arr, quoted=False))),
      side=left,
      kind=array)])
```